modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/ProposalHashSelect.scala

```diff
@@ -0,0 +1,43 @@
+package org.tessellation.sdk.infrastructure.snapshot
+
+import cats.Monad
+import cats.syntax.functor._
+
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.schema.trust.{TrustScores, defaultPeerTrustScore}
+import org.tessellation.sdk.domain.snapshot.ProposalSelect
+import org.tessellation.sdk.domain.snapshot.ProposalSelect.occuranceTrustScoreOrdering
+import org.tessellation.sdk.infrastructure.consensus.PeerDeclarations
+import org.tessellation.sdk.infrastructure.snapshot.WeightedSampling.WeightRefinement
+
+import eu.timepit.refined.numeric.Positive
+import eu.timepit.refined.refineV
+
+object ProposalHashSelect {
+
+  def make[F[_]: Monad](
+    getCurrentTrust: F[TrustScores]
+  ): ProposalSelect[F] = (declarations: Map[PeerId, PeerDeclarations]) =>
+    getCurrentTrust.map(_.scores).map { trustScores =>
+      declarations.view
+        .mapValues(_.proposal)
+        .toList
+        .collect {
+          case (peerId, Some(proposal)) =>
+            val score = trustScores.getOrElse(peerId, defaultPeerTrustScore.value)
+
+            proposal.hash -> refineV[WeightRefinement](score)
+        }
+        .collect { case (hash, Right(score)) => hash -> score.value }
+        .groupMapReduce { case (hash, _) => hash } { case (_, score) => score }(
+          _ + _
+        )
+        .view
+        .mapValues(refineV[Positive](_))
+        .collect {
+          case (hash, Right(score)) => hash -> score
+        }
```

Comment:
Could we not use the unsafe method here since we're guaranteed the value will be positive? E.g. Here's a possible alternate implementation that also makes the newly created _Ordering_ unnecessary.
```
declarations.map {
  case (peerId, pd) =>
    pd.proposal
      .map(_.hash -> trustScores.getOrElse(peerId, defaultPeerTrustScore.value))
      .filter(_._2 > 0.0)
}
  .flatten
  .groupMapReduce { case (hash, _) => hash } { case (_, score) => score }(_ + _)
  .toList
  .sortBy(_._2)(Ordering[Double].reverse)
  .map { case (h, d) => h -> Refined.unsafeApply(d) }
```

Reply:
Changed to

```
    getCurrentTrust.map(_.scores).map { trustScores =>
      declarations.map {
        case (peerId, declarations) =>
          declarations.proposal.map {
            _.hash -> trustScores.getOrElse(peerId, defaultPeerTrustScore.value)
          }.filter { case (_, s) => s > 0.0 }
      }.flatten.groupMapReduce { case (hash, _) => hash } { case (_, score) => score }(_ + _).toList.sortBy { case (_, score) => score }(
        Order[Double].toOrdering.reverse
      ).map { case (h, s) => h -> Refined.unsafeApply(s) }
```

Reply:
Can the formatting be cleaned up a bit? I find this harder to read than the sample I provided.

Reply:
What do you recommend?

Reply:
Per our offline discussion, this is no longer relevant.